Core/Src/pdu.c

```diff
@@ -5,21 +5,61 @@
 #include <stdio.h>
 #include <stdlib.h>
 
-#define PUMP_CTRL      0
-#define RADFAN_CTRL    1
-#define BRKLIGHT_CTRL  2
-#define BATBOXFAN_CTRL 3
-#define RTDS_CTRL      7 // PORT 17 BANK 1 (so read with 1_REG)
+#define PUMP_CTRL	0
+#define MPU_FAULT	2
+#define BRKLIGHT_CTRL	3
+#define FANBATTBOX_CTRL 4
+#define RTDS_CTRL	7 // PORT 17 BANK 1 (so read with 1_REG)
+// #define RADFAN_CTRL    1
 // #define TSMS_CTRL	   0x04
 // #define SMBALERT	   0x05
 #define MUTEX_TIMEOUT osWaitForever /* ms */
 
-#define SHUTDOWN_ADDR PCA_I2C_ADDR_3
-#define CTRL_ADDR     PCA_I2C_ADDR_2
+#define SHUTDOWN_ADDR PCA_I2C_ADDR_1
+#define CTRL_ADDR     PCA_I2C_ADDR_0
 #define RTDS_DURATION 1750 /* ms at 1kHz tick rate */
 
+#define MOTOR_CONTROLLER_CURRENT_SENSOR_ADDR 0x40
+#define BATTBOX_FANS_CURRENT_SENSOR_ADDR     0x42
+#define PUMPS_CURRENT_SENSOR_ADDR	     0x44
+#define LV_BOARDS_CURRENT_SENSOR_ADDR	     0x45
+
 static osMutexAttr_t pdu_mutex_attributes;
 
+extern I2C_HandleTypeDef hi2c2;
+
+// Wrapper for reading ina226 (current sensor) registers
+static inline int ina_read_reg(uint16_t dev_addr, uint8_t reg, uint16_t *data)
+{
+	uint8_t buff[2];
+	HAL_StatusTypeDef status;
+
+	status = HAL_I2C_Mem_Read(&hi2c2, dev_addr, reg, I2C_MEMADD_SIZE_16BIT,
+				  buff, 2, HAL_MAX_DELAY);
+	if (status != HAL_OK) {
+		return -1;
+	}
+
+	*data = (buff[0] << 8) | buff[1];
+	return 0;
+}
+
+// Wrapper for writing ina226 (current sensor) registers
+static inline int ina_write_reg(uint16_t dev_addr, uint8_t reg, uint16_t *data)
+{
+	uint8_t buff[2];
+	HAL_StatusTypeDef status;
+
+	status = HAL_I2C_Mem_Write(&hi2c2, dev_addr, reg, I2C_MEMADD_SIZE_16BIT,
+				   buff, 2, HAL_MAX_DELAY);
+	if (status != HAL_OK) {
+		return -1;
+	}
+
+	*data = (buff[0] << 8) | buff[1];
```

Comment:
what is this line for?

Reply:
I just looked and the whole ina_write_reg wrapper function was messed up (I prolly just copied down ina_read_reg and then forgot). It's fixed now though